prometheus/client.go

```diff
@@ -14,14 +15,33 @@ import (
 	"github.com/swift-sunshine/swscore/config"
 )
 
-const istio_request_query = "istio_request_count{destination_service=~\"%s.%s.*\"}"
-
 // Client for Prometheus API.
 // It hides the way we query Prometheus offering a layer with a high level defined API.
 type Client struct {
 	p8s api.Client
 }
 
+// Metric is a base interface for either MetricValue (single scalar, gauge or counter) or MetricHistogram
```

Comment:
Is it the place for model types or should we have a dedicated package?

Reply:
I put in the client, the wrapper for the connection and basic ones, if it grows as it happens in kubernetes/istio I separated the files.